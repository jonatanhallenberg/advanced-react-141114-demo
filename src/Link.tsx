type LinkProps = {
    url: string;
    children: React.ReactNode;
}

export const Link = ({ url, childrenÂ }: LinkProps) => (
    <a href={url} className="mylink">{children}</a>
)