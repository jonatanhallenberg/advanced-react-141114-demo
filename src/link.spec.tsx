import { render, screen } from '@testing-library/react';
import { Link } from './Link';
import '@testing-library/jest-dom';

test('should render link text', () => {
    render(<Link url="http://www.google.se">click here123</Link>)
    const linkElement = screen.getByText('click here');
    expect(linkElement).toBeInTheDocument();
})

test('should render according to snapshot', () => {
    const { asFragment } = render(<Link url="http://www.google.se">click here</Link>);
    expect(asFragment()).toMatchSnapshot();
})