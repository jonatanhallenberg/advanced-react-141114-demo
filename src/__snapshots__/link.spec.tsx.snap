// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render according to snapshot 1`] = `
<DocumentFragment>
  <a
    class="mylink"
    href="http://www.google.se"
  >
    click here
  </a>
</DocumentFragment>
`;
